services/packagist/packagist-license.service.js

```diff
@@ -3,19 +3,16 @@
 const Joi = require('joi')
 const { renderLicenseBadge } = require('../licenses')
 const { optionalUrl } = require('../validators')
-const { NotFound } = require('..')
 const {
   keywords,
   BasePackagistService,
   customServerDocumentationFragment,
 } = require('./packagist-base')
 
-const packageSchema = Joi.object()
-  .pattern(
-    /^/,
+const packageSchema = Joi.array()
+  .items(
     Joi.object({
-      'default-branch': Joi.bool(),
-      license: Joi.array().required(),
+      license: Joi.array(),
```

Comment:
Is there a need to drop `.required()` here? Is this because the new schema may not always define the key? If that's the case then I think in the transform function we need to account for that fact and throw a not found/invalid response error

Reply:
The new api endpoint now only returns the license array with the first item in the response (most recent release). At least that's what I've found in my testing. because of this, the schema validation would fail when the response contains more than 1 version. I do think that throwing a not found error would be good just in case there isn't a license at all for some reason. I'll add that in shortly.

Reply:
I think ideally you would run the `MetadataMinifier::expand` logic on the response *before* validating the schema, and then you can restore the `required()` I'd say. Note that packagist.org does not enforce the presence of a license key, so it may well be missing if the package author did not include it. I'd not expect they'd display a shields.io license badge if they haven't defined a license though but who knows :)

Reply:
@Seldaek i thought about that too but I don’t think it would be possible with the way Shields makes a request since it’s shared with other services. Maybe with some sort of event hook system but that seems like overkill for this one situation.